gnuradio-runtime/include/gnuradio/basic_block.h

```diff
@@ -28,6 +28,21 @@
 
 namespace gr {
 
+class msg_queue_comparator
+{
+public:
+    bool operator()(pmt::pmt_t const& p1, pmt::pmt_t const& p2) const
+    {
+        std::string s1 = pmt::symbol_to_string(p1);
+        std::string s2 = pmt::symbol_to_string(p2);
+        if (s1 == "system")
+            s1 = "";
+        if (s2 == "system")
+            s2 = "";
+        return s1 < s2;
```

Comment:
First off, message queues should always be identified via pmts. cf. https://github.com/gnuradio/gnuradio/blob/fa8953de2ae67990ccb052af08affac073e106ae/gnuradio-runtime/lib/basic_block.cc#L76
and the following methods. Also, `is_symbol()` requires that the PMT is a string.
https://github.com/gnuradio/gnuradio/blob/fa8953de2ae67990ccb052af08affac073e106ae/gnuradio-runtime/lib/pmt/pmt_int.h#L44

I suggest to add a private member to `msg_queue_comparator`
```c++
const pmt::pmt_t d_system_queue_key = pmt::mp("system");
```
Then you can compare the inputs `p1` and `p2` via `pmt::equal`. It looks like this is how its done in the `pmt` implementation. It might avoid some corner cases. In case it is not the system queue, I suggest to fall back to `pmt::comparator`.